jupyter-book/introduction/raw_data_processing.md

```diff
@@ -1,52 +1,397 @@
+90 \
+splici_ref
+
+# Indexing the reference
+# salmon index -t extend_txome.fa -i idx_out_dir -p num_threads
+salmon index \
+-t $(ls splici_ref/*.fa) \
+-i splici_idx \
+-p 8 
+
+# collecting reads files
+fastq_dir="toy_read_fastq"
+reads1_pat="_R1_"
+reads2_pat="_R2_"
+reads1="$(find -L $fastq_dir -name "*$reads1_pat*" -type f | xargs | sort | awk '{print $0}')"
+reads2="$(find -L $fastq_dir -name "*$reads2_pat*" -type f | xargs | sort | awk '{print $0}')"
+
+# Mapping
+# salmon alevin -i idx_out_dir -l library_type -1 reads1_files -2 reads2_files -p num_threads -o map_out_dir
+salmon alevin \
+-i splici_idx \
+-l ISR \
+-1 ${reads1} \
+-2 ${reads2} \
+-p 8 \
+-o alevin_map \
+--chromiumV3 \
+--sketch
+```
+
+This will produce an output folder called `alevin_map` that contains all the information we need to process the mapped reads using `alevin-fry`. 
+
+(raw-proc:example-quant)=
+### Processing the mapped reads
+
+Next, we run the cell barcode correction, UMI resolution, and quantification steps described previously using `alevin-fry`. This procedure involves three `alevin-fry` commands:
+
+1. The `generate-permit-list` command is used for cell barcode correction.
+2. The `collate` command filters out invalid mapping records, corrects cell barcodes, and collates mapping records originating from the same cell.
+3. The `quant` command performs UMI resolution and quantification.
+
+```bash
+# cell barcode correction
+# alevin-fry generate-permit-list -u CB_permit_list -d expected_orientation -o gpl_out_dir 
+alevin-fry generate-permit-list \
+-u 3M-february-2018.txt \
+-d fw \
+-i alevin_map \
+-o gpl
+ 
+# filtering mapping information
+# alevin-fry collate -i gpl_out_dir -r alevin_map_dir -t num_threads
+alevin-fry collate \
+-i gpl \
+-r alevin_map \
+-t 8
+
+# UMI resolution + quantification
+# alevin-fry quant -r resolution -m txp_to_gene_mapping -i gpl_out_dir -o quant_out_dir -t num_threads
+alevin-fry quant -r cr-like \
+-m $(ls splici_ref/*3col.tsv) \
+-i gpl \
+-o quant \
+-t 8
+```
+
+After running these commands, the resulting quantification information can be found in `quant/alevin`. Within this directory, there are three files: `quants_mat.mtx`, `quants_mat_cols.txt`, and `quants_mat_rows.txt`, which correspond, respectively, to the count matrix, the gene names for each column of this matrix, and the corrected, filtered cell barcodes for each row of this matrix. Of note here is the fact that `alevin-fry` was run in USA-mode, and so quantification was performed for both the spliced and unspliced status of each gene — the resulting `quants_mat_cols.txt` fill will then have a number of rows equal to 3 times the number of annotated genes which correspond, to the names used for the spliced, unspliced, and splicing-ambiguous variants of each gene. Other relevant information concerning the mapping, CB correction, and UMI resolution steps can be found in the `alevin_map`, `gpl`, and `quant` folders, respectively.
```

Comment:
> USA-mode
Might require another half sentence? No hits for USA-mode anywhere else. Smart readers can deduce that this is for unspliced, spliced and ambiguous, but maybe be explicit

Reply:
Done.